pandas/_libs/missing.pyx

```diff
@@ -205,53 +172,19 @@ cpdef ndarray[uint8_t] isnaobj(ndarray arr):
     result = np.empty(n, dtype=np.uint8)
     for i in range(n):
         val = arr[i]
-        result[i] = checknull(val)
+        result[i] = checknull(val, inf_as_na=inf_as_na)
     return result.view(np.bool_)
 
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
 def isnaobj_old(arr: ndarray) -> ndarray:
-    """
-    Return boolean mask denoting which elements of a 1-D array are na-like,
-    defined as being any of:
-     - None
-     - nan
-     - INF
-     - NEGINF
-     - NaT
-     - NA
-     - Decimal("NaN")
-
-    Parameters
-    ----------
-    arr : ndarray
-
-    Returns
-    -------
-    result : ndarray (dtype=np.bool_)
-    """
-    cdef:
-        Py_ssize_t i, n
-        object val
-        ndarray[uint8_t] result
-
-    assert arr.ndim == 1, "'arr' must be 1-D."
-
-    n = len(arr)
-    result = np.zeros(n, dtype=np.uint8)
-    for i in range(n):
-        val = arr[i]
-        result[i] = (
-            checknull(val)
-            or util.is_float_object(val) and (val == INF or val == NEGINF)
-        )
-    return result.view(np.bool_)
+    return isnaobj(arr, inf_as_na=True)
```

Comment:
Is it possible to just call `isnaobj(arr, inf_as_na=True)` where `isnaobj_old(arr)` is called (same for isnaobj2d)?

Reply:
it is, im planning to do that in a separate pass